packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -140,7 +141,7 @@ export interface Point {
  *
  * @public
  */
-
+@moveable
```

Comment:
so it does not apply to subclasses automatically?

Reply:
would we need to mark every subclass as movable?

Reply:
We only need to mark subclasses that we want the `move` function on. Alternatively we could manually write it in each class, but using a decorator makes more sense here :P

Reply:
it means that the build time types do not match the runtime types? one can declare the base class implementing movable but actually forget to add decorators to subclasses.

I suggest that implement move in the base JSHandle class given that we don't have plan resource management for other resources so far.

Reply:
The build time types do match the runtime types. We also wrote `@moveable` on the JSHandle.

Reply:
Sorry, what I wrote was incorrect. We only need to mark _the base class_ that we want the `move` function on. The reason we do it on ElementHandle is because the inherited move is not the one we want, but the one coming from `this.handle`, so we use `@moveable` again to just create the generic one on ElementHandle.

Reply:
In general, as long as you declare it on the base class you are fine.

Reply:
But we would not need to do that if just declare mode on the base class, right? That looks like a simpler solution. Btw let's add a move test for ElementHandle too?

Reply:
We would need to sadly. See how `dispose` is implemented in `ElementHandle`.

Reply:
See my other comment (seems to work)

Reply:
Indeed it does work!